tools/c7n_mailer/c7n_mailer/utils.py

```diff
@@ -25,6 +26,35 @@ class Providers:
     GCP = 2
 
 
+def session_factory(mailer_config):
+    return boto3.Session(
+        region_name=mailer_config['region'],
+        profile_name=mailer_config.get('profile', None))
+
+
+def get_processor(mailer_config, logger):
+    """Find the appropriate queue processor for a given provider
+
+    Import provider-specific processor modules lazily. This way running or provisioning
+    the mailer only requires additional provider packages if they're used in the
+    mailer config.
+    """
+    provider = get_provider(mailer_config)
+
+    if provider == Providers.Azure:
+        from c7n_mailer.azure_mailer.azure_queue_processor import MailerAzureQueueProcessor
```

Comment:
we should wrap the gcp/azure imports in try/except with instructions on how to install with extras

Reply:
```suggestion
        try:
          from c7n_mailer.azure_mailer.azure_queue_processor import MailerAzureQueueProcessor
        except ImportError:
          raise Exception("Missing c7n-azure package, install via pip install c7n-mailer --extras azure')
```

Reply:
Good call. GCP already had a message (though it was deeper where we try to import from `c7n_gcp`). I added similar behavior for Azure, and included extras recommendations in the failure message.